src/main/java/chatbuddy/ui/Ui.java

```diff
@@ -0,0 +1,110 @@
+package chatbuddy.ui;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import chatbuddy.task.Task;
+
+/**
+ * The Ui class handles interactions with the user.
+ * It reads inputs from the users and output messages to the user.
+ */
+public class Ui {
```

Comment:
Perhaps you can consider creating an attribute to represent the buffer in front of each lines printed onto the console?

I believe that might help minimize the chance for a line printed by Ui to have a different buffer size at the front.